lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+
+The following is a comprehensive list of all expressions allowed in guards:
+
+  * comparison operators (`==`, `!=`, `===`, `!==`, `>`, `>=`, `<`, `<=`)
+  * strictly boolean operators (`and`, `or`, `not`) (the `&&`, `||`, and `!` sibling operators are not allowed as they're not *strictly* boolean - meaning they don't require both sides to be booleans)
+  * arithmetic binary operators (`+`, `-`, `*`, `/`)
+  * arithmetic unary operators (`+`, `-`)
+  * binary concatenation operator (`<>`)
+  * `in` operator (as long as the right-hand side is a list or a range)
+  * the following "type-check" functions (all documented in the `Kernel` module):
+    * `is_atom/1`
+    * `is_binary/1`
+    * `is_bitstring/1`
+    * `is_boolean/1`
+    * `is_float/1`
+    * `is_function/1`
+    * `is_function/2`
+    * `is_integer/1`
+    * `is_list/1`
+    * `is_map/1`
+    * `is_nil/1`
+    * `is_number/1`
+    * `is_pid/1`
+    * `is_port/1`
+    * `is_reference/1`
+    * `is_tuple/1`
+  * the following guard-friendly functions (all documented in the `Kernel` module):
+    * `abs/1`
+    * `binary_part/3`
+    * `bit_size/1`
+    * `byte_size/1`
+    * `div/2`
+    * `elem/2`
+    * `hd/1`
+    * `length/1`
+    * `map_size/1`
+    * `node/0`
+    * `node/1`
+    * `rem/2`
+    * `round/1`
+    * `self/0`
+    * `tl/1`
+    * `trunc/1`
+    * `tuple_size/1`
+  * the following handful of Erlang bitwise operations, if imported from the `Bitwise` module:
+    * `band/2` or the `&&&` operator
+    * `bor/2` or the `|||` operator
+    * `bnot/1` or the `~~~` operator
+    * `bsl/1` or the `<<<` operator
+    * `bsr/1` or the `>>>` operator
+    * `bxor/2` or the `^^^` operator
+
+## Defining custom guard expressions
+
+As mentioned before, only the expressions listed in the previous section are allowed in guards. However, we can take advantage of macros to write custom guards that can simplify our programs or make them more domain-specific. At the end of the day, what matters is that the *output* of the macros (which is what will be compiled) boils down to a combinations of the allowed expressions.
+
+Let's look at a quick case study: we want to check that a function argument is a keyword list (a list of zero or more `{key, value}` pairs where `key` is an atom). With pattern matching, this is impossible to do since the list could be of any length. It's impossible to check this with guards as well: we can't check *all* of the elements of the list; however, we can perform a weak check by checking that only the first element of the list is a `{atom, value}` tuple. While this does not ensure the argument is a keyword list, it may be enough to be useful in real-world scenarios.
+
+This guard would look like this:
+
+```elixir
+def is_keyword_list(list)
```

Comment:
I also don't like this example since we can't traverse the whole list and this will return false for empty lists. What if we write something to check if it is a MFA tuple (module-function-arity)?